statsmodels/sandbox/mice/mice.py

```diff
@@ -0,0 +1,1287 @@
+`combine` method.  Results and various summary plots are then
+available.
+
+Terminology
+-----------
+
+The MICE procedure is determined by a family of conditional models.
+There is one conditional model for each variable with missing values.
+A conditional model may be conditioned on all or a subset of the
+remaining variables, using main effects, transformations,
+interactions, etc. as desired.
+
+A 'perturbation method' is a method for setting the parameter estimate
+in a conditional model.  The 'gaussian' perturbation method first fits
+the model (usually using maximum likelihood, but it could use any
+statsmodels fit procedure), then sets the parameter vector equal to a
+draw from the Gaussian approximation to the sampling distribution for
+the fit.  The 'bootstrap' perturbation method sets the parameter
+vector equal to a fitted parameter vector obtained when fitting the
+conditional model to a bootstrapped version of the data set.
+
+Class structure
+---------------
+
+There are two main classes in the module:
+
+* 'MICEData' wraps a Pandas dataframe, incorporating information about
+  the conditional models for each variable with missing values. It can
+  be used to produce multiply imputed data sets that are to be further
+  processed or distributed to other researchers.  A number of plotting
+  procedures are provided to visualize the imputation results and
+  missing data patterns.  The `history_func` hook allows any features
+  of interest of the imputed data sets to be saved for further
+  analysis.
+
+* 'MICE' takes both a 'MICEData' object and an analysis model
+  specification.  It runs the multiple imputation, fits the analysis
+  models, and combines the results to produce a `MICEResults` object.
+  The summary method of this results object can be used to see the key
+  estimands and inferential quantities..
+
+Notes
+-----
+
+By default, to conserve memory 'MICEData' saves very little
+information from one iteration to the next.  The data set passed by
+the user is copied on entry, but then is over-written each time new
+imputations are produced.  If using 'MICE', the fitted
+analysis models and results are saved.  MICEData includes a
+`history_callback` hook that allows arbitrary information from the
+intermediate datasets to be saved for future use.
+
+References
+----------
+
+JL Schafer: 'Multiple Imputation: A Primer', Stat Methods Med Res,
+1999.
+
+TE Raghunathan et al.: 'A Multivariate Technique for Multiply
+Imputing Missing Values Using a Sequence of Regression Models', Survey
+Methodology, 2001.
+
+SAS Institute: 'Predictive Mean Matching Method for Monotone Missing
+Data', SAS 9.2 User's Guide, 2014.
+
+A Gelman et al.: 'Multiple Imputation with Diagnostics (mi) in R:
+Opening Windows into the Black Box', Journal of Statistical Software,
+2009.
+"""
+
+import pandas as pd
+import numpy as np
+import patsy
+import statsmodels.api as sm
+import statsmodels
```

Comment:
it's better to import from specific modules or subdirectories, than from `sm` api
I've only seen OLS so far.